pontoon/sync/formats/compare_locales.py

```diff
@@ -0,0 +1,116 @@
+"""
+Parser and serializer for file formats supported by compare-locales library:
+https://hg.mozilla.org/l10n/compare-locales/
+"""
+from __future__ import absolute_import
+
+import logging
+
+from collections import OrderedDict
+from compare_locales import (
+    parser,
+    serializer,
+)
+
+from pontoon.sync import SyncError
+from pontoon.sync.exceptions import ParseError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+
+log = logging.getLogger(__name__)
+
+
+class CompareLocalesEntity(VCSTranslation):
+    """
+    Represents an entity in a file handled by compare-locales.
+    """
+    def __init__(self, key, string, comment, order):
+        self.key = key
+        self.source_string = string
+        self.source_string_plural = ''
+        self.strings = {None: self.source_string} if self.source_string else {}
+        self.comments = [comment.raw_val] if comment else []
+        self.order = order
+        self.fuzzy = False
+        self.source = []
+
+
+class CompareLocalesResource(ParsedResource):
+    def __init__(self, path, source_resource=None):
+        self.path = path
+        self.entities = OrderedDict()  # Preserve entity order.
+        self.source_resource = source_resource
+        self.parser = parser.getParser(self.path)
+
+        # A monolingual l10n file might not contain all entities, but the code
+        # expects ParsedResource to contain representations of all of them. So
+        # when parsing the l10n resource, we first create empty entity for each
+        # source resource entity.
+        if source_resource:
+            for key, entity in source_resource.entities.items():
+                self.entities[key] = CompareLocalesEntity(
+                    entity.key,
+                    None,
+                    None,
+                    None,
+                )
+
+        try:
+            self.parser.readFile(self.path)
+        except IOError as err:
+            # If the file doesn't exist, but we have a source resource,
+            # we can keep going, we'll just not have any translations.
+            if source_resource:
+                return
+            else:
+                raise ParseError(err)
+
+        entities = list(self.parser.parse())
+        for order, entity in enumerate(entities):
+            if isinstance(entity, parser.Entity):
```

Comment:
We need this check in case the file does not exists. In which case the parser returns `Junk`.